Foundation/NSOrderedSet.swift

```diff
@@ -46,85 +49,81 @@ open class NSOrderedSet : NSObject, NSCopying, NSMutableCopying, NSSecureCoding,
         }
     }
     
-    public required convenience init?(coder aDecoder: NSCoder) {
+    public required init?(coder aDecoder: NSCoder) {
         guard aDecoder.allowsKeyedCoding else {
             preconditionFailure("Unkeyed coding is unsupported.")
         }
+        _orderedStorage = NSMutableArray()
+        _storage = NSMutableSet()
         var idx = 0
-        var objects : [AnyObject] = []
         while aDecoder.containsValue(forKey: ("NS.object.\(idx)")) {
             guard let object = aDecoder.decodeObject(forKey: "NS.object.\(idx)") else {
                 return nil
             }
-            objects.append(object as! NSObject)
+            _orderedStorage.add(object)
+            _storage.add(object)
             idx += 1
         }
-        self.init(array: objects)
```

Comment:
the previous implementation is a little bit more correct I think.  You don't want to have multiple initializers for coding. Coding should use the regular initializer as much as possible -- philosophically:  in doing so its easier to maintain the run time invariants of your object, since they're automatically enforced. 

Also from a correctness perspective -- its possible to construct a payload that will not actually come from a valid runtime ordered set (e.g. contain duplicates) and this does not detect such a malformed archive.  Best approach there would be just create the ordered set and ask if its count == count of the array, returning nil if not equal.

Reply:
Thank you for the insight, I'll take another look and see if I can get it closer to was it was before